src/libclang/libclang_parser.cpp

```diff
@@ -120,6 +120,9 @@ std::string get_full_path(const detail::cxstring& dir, const std::string& file)
     if (is_absolute(file))
         // absolute file
         return file;
+    if (dir.empty())
+        // should never happen, but GCC 12.2 is not convinced. That's what exceptions are for, isn't it?
+        throw std::invalid_argument("Either file is absolute or relative to some named dir");
```

Comment:
No, that's what assertions are for ;)

Please assert instead at the beginning of the function.

Reply:
Assertion was not enough, so I had to place a `#pragma`. Not as beautiful but at least one can easily spot that the line would be unreachable if `dir` was empty (unless clever users decides to disable the asserts, then they would be on their own luck :) )